index.js

```diff
@@ -86,8 +86,23 @@ module.exports = function (str, opts) {
 		delete urlObj.search;
 	}
 
+	var queryParameters = queryString.parse(urlObj.search);
+
+	// remove query unwanted parameters
+	if (Array.isArray(opts.ignoredQueryParameters)) {
+		for (var key in queryParameters) {
+			if (!queryParameters.hasOwnProperty(key)) {
```

Comment:
@SamVerschueren I agree, but I would keep it as is so it's easier to convert to a `for of` loop in the future. Can just remove the `hasOwnProperty` when we upgrade to the new query-string.